src/components/routerView.vue

```diff
@@ -43,21 +46,27 @@
       return null
     }
 
-    const components = getComponents(match)
-    const component = components[name]
+    const component = getComponent(match)
+    const props = getProps(match)
 
-    if (component) {
-      if (typeof component === 'function') {
-        return defineAsyncComponent(component as AsyncComponentLoader)
-      }
+    if (!component) {
+      return null
+    }
 
-      return component
+    if (props) {
+      return componentUtil(component, () => props(route.params))
```

Comment:
This is the important piece of implementation. If we detect a route had props for the component we wrap the component in our `component` utility which already handles binding props from a callback (which can return a promise).